### PR TITLE
Make sign_in_tokens expire after 30 mins by default

### PR DESCRIPTION
**Pull request by aldavidson**
### Context

See [Trello card](https://trello.com/c/3tRtyZrZ/121-make-signintokens-time-limited)

### Changes proposed in this pull request

Make sign_in_tokens expire after 10 mins by default. 
Add option to set a different expiry period via env var
Add spec for expiry/non-expiry behaviour

### Guidance to review

1. Generate a sign-in token by clicking 'Sign in' and entering your email address. Copy the sign-in URL from the footer
2. From the console, check that `sign_in_token_expires_at` is set correctly (`User.where(email_address: '...').sign_in_token_expires_at`)
3. Wait until that expires (or update it from the console), and then try and use the sign-in token URL - it should not work

4.

### CODE REVIEW

**Comment by fofr** [comment]
10 minutes feels a bit short to me.

**Comment by aldavidson** [comment]
> 10 minutes feels a bit short to me.

It's just a default - the main point of this PR was to stop them having infinite lifetime. Once the code change is in, the TTL can be configured in seconds via the env var

File: app/concerns/sign_in_with_token.rb [hunk 1]
```diff
@@ -4,13 +4,26 @@ module SignInWithToken
   included do
     validates :sign_in_token, uniqueness: true, allow_nil: true
 
-    def generate_token!
-      update!(sign_in_token: SecureRandom.uuid)
+    def generate_token!(ttl: nil)
+      ttl ||= (ENV['SIGN_IN_TOKEN_TTL_SECONDS'] || 1800)
```
Comment (benilovj) [inline app/concerns/sign_in_with_token.rb:8]: Can we please pull out 1800 as a constant at the top of the file? it's a bit buried here.

Comment (aldavidson) [reply app/concerns/sign_in_with_token.rb:8]: Yep, fair point - I've done that and also extracted the calculation of effective TTL out to a separate method

**Pull request merged**